src/items/functions.md

```diff
@@ -189,6 +192,57 @@ Exhaustive list of permitted structures in const functions:
   the following unsafe operations:
     * calls to const unsafe functions
 
+## Async functions
+
+Functions may be qualified as async, and this can also be combined with the
+`unsafe` qualifier:
+
+```rust,edition2018
+async fn regular_example() { }
+async unsafe fn unsafe_example() { }
+```
+
+Async functions do no work when called: instead, they
+capture their arguments into a future. When polled, that future will
+execute the function's body.
+
+An async function is roughly equivalent to a function
+that returns [`impl Future`] and with an [`async move` block][async-blocks] as
+its body:
+
+```rust,edition2018
```

Comment:
I think we'll need to add `,ignore` here.  This won't compile because the identifier `example` is used twice.  I can't really think of a nice way to make it work better. I don't like using `ignore` because examples often end up broken, but I think changing the identifier would make it less clear.

Reply:
Maybe just two blocks, like so:

---

```rust
// Source
async fn example(x: &str) -> usize {
    x.len()
}
```

is roughly equivalent to 

```rust
// Desugared
fn example<'a>(x: &'a str) -> impl Future<Output = usize> + 'a {
    async move { x.len() }
}
```

Reply:
I think that would be good, too.  If you do that, be sure to add:
`# use std::future::Future;` so it can compile.